lib/house.rb

```diff
@@ -0,0 +1,44 @@
+class House
+  attr_reader :price, :address, :rooms, :details
+
+  def initialize(price, address)
+    # I don't think it's necessary to preserve anything other than fixnums in this attribute.
+    # Regexp superior to just deleting the $ to account for any other invalid character
+    @price = price.gsub(/[^0-9]/, "").to_i
+    @address = address
+    @rooms = []
+    @details = {"price"=>@price, "address"=>@address}
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_range?
```

Comment:
Remember to name methods exactly what they are in the spec. Otherwise, they won't pass our internal tests we use to grade these.